File: kw_email/result.py

class Result:
    """
     * Result from sending service
    """

    def __init__(self, status: bool = False, data: str = None, remote_id: str = None):
        self.status = status
        self.data = data
        self.remote_id = remote_id

    def sanitize(self):
        self.status = bool(self.status) if self.status is not None else False
        self.data = str(self.data) if self.data is not None else None
        self.remote_id = str(self.remote_id) if self.remote_id is not None else None

    def get_status(self) -> bool:
        return bool(self.status)

    def get_data(self) -> str:
        return self.data

    def get_remote_id(self) -> str:
        return self.remote_id

File: php-src/Basics/Result.php
<?php

namespace kalanis\EmailApi\Basics;


/**
 * Class Result
 * @package kalanis\EmailApi\Basics
 * Result from sending service
 */
class Result
{
    public bool $status = false;
    public ?string $data = null;
    public ?string $remoteId = null;

    public function __construct(bool $status = false, ?string $data = null, ?string $remoteId = null)
    {
        $this->status = $status;
        $this->data = $data;
        $this->remoteId = $remoteId;
    }

    public function getStatus(): bool
    {
        return $this->status;
    }

    public function getData(): ?string
    {
        return $this->data;
    }

    public function getRemoteId(): ?string
    {
        return $this->remoteId;
    }
}
